src/devices/controller/controller.cpp

```diff
@@ -109,7 +146,196 @@ void loop() {
+    } else {
+      set_effect_packet.writeSetEffect(/*effect_index=*/0, /*delay=*/1,
+                                       /*palette_index=*/palettes[i / 6]);
+      uint8_t led_index = (i % 6);
+      SetMainLed(
+          i, palette_effect.GetRGB(led_index, state_machine.GetNetworkMillis(),
+                                   kPaletteStrip, &set_effect_packet));
+    }
+  }
+
+  bool pressed = false;
+  uint8_t palette_index = 0;
+  if (left_buttons.Button1Pressed()) {
+    pressed = true;
+    palette_index = 0;
+  } else if (left_buttons.Button2Pressed()) {
+    pressed = true;
+    palette_index = 2;
+  } else if (left_buttons.Button3Pressed()) {
+    pressed = true;
+    palette_index = 4;
+  } else if (right_buttons.Button1Pressed()) {
+    pressed = true;
+    palette_index = 1;
+  } else if (right_buttons.Button2Pressed()) {
+    pressed = true;
+    palette_index = 3;
+  } else if (right_buttons.Button3Pressed()) {
+    pressed = true;
+    palette_index = 5;
+  }
+
+  if (left_buttons.Button1Pressed()) {
+    SetLeftButtonLeds(kButtonPressedBrightness, kButtonActiveBrightness,
+                      kButtonActiveBrightness);
+  } else if (left_buttons.Button2Pressed()) {
+    SetLeftButtonLeds(kButtonActiveBrightness, kButtonPressedBrightness,
+                      kButtonActiveBrightness);
+  } else if (left_buttons.Button3Pressed()) {
+    SetLeftButtonLeds(kButtonActiveBrightness, kButtonActiveBrightness,
+                      kButtonPressedBrightness);
+  } else {
+    SetLeftButtonLeds(kButtonActiveBrightness, kButtonActiveBrightness,
+                      kButtonActiveBrightness);
+  }
+
+  if (right_buttons.Button1Pressed()) {
+    SetRightButtonLeds(kButtonPressedBrightness, kButtonActiveBrightness,
+                       kButtonActiveBrightness);
+  } else if (left_buttons.Button2Pressed()) {
+    SetRightButtonLeds(kButtonActiveBrightness, kButtonPressedBrightness,
+                       kButtonActiveBrightness);
+  } else if (left_buttons.Button3Pressed()) {
+    SetRightButtonLeds(kButtonActiveBrightness, kButtonActiveBrightness,
+                       kButtonPressedBrightness);
+  } else {
+    SetRightButtonLeds(kButtonActiveBrightness, kButtonActiveBrightness,
+                       kButtonActiveBrightness);
+  }
+
+  SetBottomButtonLeds(0, 0, 0);
+
+  if (pressed) {
+    set_effect_packet.writeSetEffect(state_machine.GetEffectIndex(),
+                                     kSetEffectDelay, palettes[palette_index]);
+    state_machine.SetEffect(&set_effect_packet);
+  }
+}
+
+void setup() {
+  FastLED.addLeds<NEOPIXEL, kNeopixelPin>(leds.data(), kLedCount)
+      .setCorrection(TypicalLEDStrip);
+  FastLED.setBrightness(32);
+
+  if (!radio.Begin()) {
+    while (true) {
+      leds[0] = CRGB((millis() / 200) % 2 == 0 ? 255 : 0, 0, 0);
+      FastLED.show();
+    }
+  }
+  FastLED.clear(/*writeData=*/true);
+}
+
+void loop() {
+  state_machine.Tick();
+
+  left_buttons.Tick();
```

Comment:
Nit: (for follow up) I think I like "flush" better than "Tick" if all these are doing is flushing the state to the LED strip

Reply:
Acknowledged, will update in a follow-up.